packages/uploadthing/src/sdk/index.ts

```diff
@@ -3,319 +3,288 @@ import { generateUploadThingURL, UploadThingError } from "@uploadthing/shared";
- * @param {string | string[]} fileKeys
- *
- * @example
- * await deleteFiles("2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg");
- *
- * @example
- * await deleteFiles(["2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg","1649353b-04ea-48a2-9db7-31de7f562c8d_image2.jpg"])
- */
-export const deleteFiles = async (fileKeys: string[] | string) => {
-  guardServerOnly();
-  incompatibleNodeGuard();
-
-  if (!Array.isArray(fileKeys)) fileKeys = [fileKeys];
-
-  const res = await fetch(generateUploadThingURL("/api/deleteFile"), {
-    method: "POST",
-    cache: "no-store",
-    headers: {
+  constructor(opts?: UTApiOptions) {
+    this.fetch = opts?.fetch ?? globalThis.fetch;
+    this.apiKey = opts?.apiKey ?? process.env.UPLOADTHING_SECRET;
+    this.defaultHeaders = {
       "Content-Type": "application/json",
-      "x-uploadthing-api-key": getApiKeyOrThrow(),
+      "x-uploadthing-api-key": this.apiKey!,
       "x-uploadthing-version": UPLOADTHING_VERSION,
-    },
-    body: JSON.stringify({ fileKeys }),
-  });
-
-  if (!res.ok) {
-    throw new Error("Failed to delete files");
+    };
   }
 
-  return res.json() as Promise<{ success: boolean }>;
-};
-
-/**
- * Request file URLs from UploadThing storage.
- * @param {string | string[]} fileKeys
- *
- * @example
- * const data = await getFileUrls("2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg");
- * console.log(data); // [{key: "2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg", url: "https://uploadthing.com/f/2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg"}]
- *
- * @example
- * const data = await getFileUrls(["2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg","1649353b-04ea-48a2-9db7-31de7f562c8d_image2.jpg"])
- * console.log(data) // [{key: "2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg", url: "https://uploadthing.com/f/2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg" },{key: "1649353b-04ea-48a2-9db7-31de7f562c8d_image2.jpg", url: "https://uploadthing.com/f/1649353b-04ea-48a2-9db7-31de7f562c8d_image2.jpg"}]
- */
-export const getFileUrls = async (fileKeys: string[] | string) => {
-  guardServerOnly();
-  incompatibleNodeGuard();
-
-  if (!Array.isArray(fileKeys)) fileKeys = [fileKeys];
-
-  const res = await fetch(generateUploadThingURL("/api/getFileUrl"), {
-    method: "POST",
-    cache: "no-store",
-    headers: {
-      "Content-Type": "application/json",
-      "x-uploadthing-api-key": getApiKeyOrThrow(),
-      "x-uploadthing-version": UPLOADTHING_VERSION,
-    },
-    body: JSON.stringify({ fileKeys }),
-  });
+  private async requestUploadThing<T extends Record<string, unknown>>(
+    pathname: `/${string}`,
+    body: Record<string, unknown>,
+    fallbackErrorMessage: string,
+  ) {
+    // Force API key to be set before requesting.
+    // Ideally we'd just throw in the constructor but since we need to export
+    // a `utapi` object we can't throw in the constructor because it would
+    // be a breaking change.
+    // FIXME: In next major
```

Comment:
This would be a breaking change since previously this check was made when making the request, so they could still import `utapi` without a secret set and not get errors.

Not sure what we want to do to fix this, maybe just exporting the class and they initialize the `utapi` object themselves? That way you can still import from this file (or /server) without immediately throwing if you don't have a secret set in your envs

Reply:
> maybe just exporting the class and they initialize the utapi object themselves

This is how I would probably do it. It will be a breaking change, but any other way would likely still be breaking, and also almost guaranteed to be more complicated.

Reply:
> without immediately throwing if you don't have a secret set in your envs

Why do we care if this happens though? You can't use these functions without a secret in your env vars, so why not throw until you do? As long as the error message is clear 🤷

Reply:
fair enough 😂